robot/packages/cloud_package_manager.go

```diff
@@ -181,17 +133,22 @@ func (m *cloudManager) Sync(ctx context.Context, packages []config.PackageConfig
 		if ok {
 			if existing.thePackage.Package == p.Package && existing.thePackage.Version == p.Version {
 				m.logger.Debug("  Package already managed, skipping")
-
 				newManagedPackages[PackageName(p.Name)] = existing
-				delete(m.managedPackages, PackageName(p.Name))
```

Comment:
99% sure that this did nothing. Happy to revert if we just don't want to change this

Reply:
Interesting... I tried to think of cases where this is needed (two config entries, etc). I was thinking that maybe was an optimization for other functions in this file to run async, but we have a lock in place. Fine to remove as the list gets updated with correct entries at the end.